### DISCUSSION TITLE
Changing Meshes

### DISCUSSION THREAD

**Discussion by alanaraujoeu** [Sixze/ALS-Refactored#507] [General]
Hey guys, there is any tutorial showing what´s the best practices to use ALS-F? 

I tried to change Als_Character with my mesh made on CharacterCretor 4, but crashed. Using B_Als_Character instead, it works, but the eyes and jaw or glasses don´t follow the head when it look to it´s sides...

Also, my new skeleton mesh that I´m using is flying a little bit (doesn´t matter if I move it in blueprint B_Als_Character ) and the objects don´t attach correctly to it and go over the floor...

**Comment by kamilbeben** [comment]
Not sure what's the best practices yet, but recently I've started playing around with ALS myself and so far I have two ideas:

If your mesh is rigged to the default UE4 skeleton, you can use leader pose (as Als skeleton is really UE4 skeleton with some extra bones and sockets):
 - set `Visible` to `false` and `Visibility Based Anim Tick Option` to `AlwaysTickPoseAndRefreshBones`
 - Add new skeletal mesh component as a child of the default character Mesh
 - Call `Set Leader Pose Component` function with `New leader bone component` as a `target` and your new skeletal mesh component as a `Target`

If your mesh is not rigged to the default UE4 skeleton, you can use runtime retargeting:
 - Create IK Retargeter from ALS skeleton to your custom skeleton
 - Create ABP of your skeleton with a single node called `Retarget pose from mesh`
 - Set `Use attached parent` to `true` and assign previously created retargeter
 - Add new skeletal mesh component as a child of the default character Mesh
 - Assign previously created ABP to the new skeletal mesh component

I'm curious as to what someone more experienced with ALS framework would say myself, so please correct me if I'm wrong.
